nebu/cli/assemble.py

```diff
@@ -0,0 +1,71 @@
+import os
+import shutil
+from pathlib import Path
+
+import click
+from cnxepub.formatters import (
+    HTMLFormatter,
+    SingleHTMLFormatter,
+)
+from cnxepub.models import flatten_to_documents
+
+from ._common import common_params, logger
+from ..models.binder import Binder
+from ..models.utils import scan_for_id_mapping
+from ..utils import relative_path
+
+
+def produce_collection_xhtml(binder, output_dir):
+    collection_xhtml = output_dir / 'collection.assembled.xhtml'
+    with collection_xhtml.open('wb') as fb:
+        fb.write(bytes(SingleHTMLFormatter(binder)))
+
+    return collection_xhtml
+
+
+def provide_supporting_files(input_dir, output_dir, binder):
+    documents = {doc.id: doc for doc in flatten_to_documents(binder)}
+    for id, filepath in scan_for_id_mapping(input_dir).items():
+        (output_dir / id).symlink_to(relative_path(filepath.parent, output_dir))
```

Comment:
There is likely a deeper discussion to be had, but it could be a good idea to enforce a cache. This would be similar to what `pip` does with its download cache, except it'd be more like a data storage for us.

Taking this approach could hit two birds with one stone. Bird one, this baking build directory might have centrally located data source (i.e. something like `~/.data/neb/...`). Bird two, the tree format of `neb get` would be symlinks to data in this central data location. It's storing the source data separate from the actions. It's not exactly ideal, but since we've gone down this path of re-inventing version control, why not continue by differentiating the source from the changes. 😅

O... and the bonus third and forth birds would be downloads would be quicker and we could fix the problem with results being written to the output directory as they are ready.

More discussion on this is needed of course. I'm saying part of my thoughts on the matter here.